src/inference/dynamichmc.jl

```diff
@@ -52,7 +52,7 @@ function sample(model::Model,
     vi = VarInfo()
     model(vi, SampleFromUniform())
 
-    if spl.selector == DEFAULT_SELECTOR
+    if spl.parent == SampleFromPrior()
```

Comment:
@xukai92 I'm confused by why we need `spl.selector == DEFAULT_SELECTOR` here. Can you clarify?

Reply:
Perhaps we can introduce a field to `Selector` to indicate sampling algorithm types, e.g.

```julia
struct Selector
    gid :: UInt64
    alg :: Symbol # :default, :invalid, :Gibbs, :HMC
end
Selector() = Selector(time_ns(), :default)
Selector(a::Symbol) = Selector(time_ns(), a)
==(s1::Selector, s2::Selector) = s1.gid == s2.gid
```

Then we can 

- `spl.selector == DEFAULT_SELECTOR` ==> `spl.selector.alg==:default`
- Remove `INVALID_SELECTOR ` and change ` push!(vi, vn, r, dist, INVALID_SELECTOR)` ==> 
`push!(vi, vn, r, dist, Selector(:invalid))`
https://github.com/TuringLang/Turing.jl/blob/b7ceac22310175f0651f70ac150aaf3e9fad5f78/src/inference/pgibbs.jl#L173

Reply:
Good solution, I will have a try.